src/backend/src/lib.rs

```diff
@@ -398,6 +406,30 @@ fn add_user_token(token: UserToken) {
     mutate_state(|s| add_to_user_token(stored_principal, &mut s.user_token, &token, &find));
 }
 
+#[update(guard = "caller_is_not_anonymous")]
+fn set_many_user_tokens(tokens: Vec<UserToken>) {
+    let stored_principal = StoredPrincipal(ic_cdk::caller());
+
+    mutate_state(|s| {
+        for token in tokens {
+            assert_token_symbol_length(&token).unwrap_or_else(|e| ic_cdk::trap(&e));
+            assert_token_enabled_is_some(&token).unwrap_or_else(|e| ic_cdk::trap(&e));
+
+            let addr = parse_eth_address(&token.contract_address);
+
+            let find = |t: &UserToken| {
+                t.chain_id == token.chain_id && parse_eth_address(&t.contract_address) == addr
```

Comment:
`parse_eth_address` validate the address format as well. I guess it's a shortcut to make both the validation and comparison in a single line of code instead of two steps.

Note this was not introduced in this PR.

Reply:
Indeed, but it saves a line at the cost of making the code slower.  If making the code faster takes a lot of effort, it's not worth it.  I am not advocating a huge amount of pre-optimization.  But here the simpler approach is also the faster one.

Reply:
Sure not against. Can you take care of it?

Reply:
Sure thing.  https://github.com/dfinity/oisy-wallet/pull/1642/files